src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -379,6 +423,102 @@ public static string GetModifiedJsonString(string[] reps, string enumString)
+        private static string _configWithEnvVar = @"
+          {
+            ""$schema"": ""/dab-schema.json"",
+            ""data-source"": {
+              ""database-type"": ""cosmosdb_nosql"",
+              ""options"": {
+                ""database"": ""@env('DATABASE_NAME')"",
+                ""schema"": ""@env('GRAPHQL_SCHEMA_PATH')""
+              },
+              ""connection-string"": ""@env('DATABASE_CONNECTION_STRING')""
+            },
+            ""runtime"": {
+              ""rest"": {
+                ""enabled"": true,
+                ""path"": ""/api""
+              },
+              ""graphql"": {
+                ""allow-introspection"": true,
+                ""enabled"": true,
+                ""path"": ""/graphql""
+              },
+              ""host"": {
+                ""mode"": ""development"",
+                ""cors"": {
+                  ""origins"": [],
+                  ""allow-credentials"": false
+                },
+                ""authentication"": {
+                  ""provider"": ""AzureAD"",
+                  ""jwt"": {
+                    ""audience"": ""@env('JWT_AUDIENCE')"",
+                    ""issuer"": ""@env('JWT_ISSUER')""
+                  }
+                }
+              }
+            },
+            ""entities"": {
+              ""Book"": {
+                ""source"": ""@env('ENTITY_SOURCE')"",
+                ""rest"": ""@env('ENTITY_REST_PATH')"",
+                ""permissions"": [
+                    {
+                      ""role"": ""@env('ENTITY_ROLE')"",
+                      ""actions"": [ ""*"" ]
+                    }
+                ]
+              }
+            }
+          }
+        ";
+
+        /// <summary>
+        /// Clears the environment variables defined in the provided text.
+        /// </summary>
+        /// <param name="text">A string containing environment variables in the format "KEY=VALUE", one per line.</param>
+        private static void ClearEnvironmentVariablesFromText(string text)
+        {
+            string[] lines = text.Split('\n');
+            foreach (string line in lines)
+            {
+                string[] parts = line.Split('=');
+                Environment.SetEnvironmentVariable(parts[0], null);
+            }
+        }
+
+        /// <summary>
+        /// A string representing environment variables for testing environment variable replacement in the config.
+        /// </summary>
+        private static string _environmentFileContent =
+            "DATABASE_TYPE=cosmosdb_nosql" + "\n" +
+            "DATABASE_NAME=planet" + "\n" +
+            "GRAPHQL_SCHEMA_PATH=gql-schema.gql" + "\n" +
+            "DATABASE_CONNECTION_STRING=TEST_CONNECTION_STRING" + "\n" +
+            "JWT_AUDIENCE=TEST_AUDIENCE" + "\n" +
+            "JWT_ISSUER=TEST_ISSUER" + "\n" +
+            "ENTITY_SOURCE=books" + "\n" +
+            "ENTITY_ROLE=anonymous" + "\n" +
+            "ENTITY_REST_PATH=/books";
```

Comment:
why not create a dictionary so you can avoid all the helpers which parse through this text?